src/backend/distributed/planner/multi_router_planner.c

```diff
@@ -970,6 +1016,23 @@ ErrorIfModifyQueryNotSupported(Query *queryTree)
 		   commandType == CMD_DELETE);
 
 	/*
+	 * Result relation is expected to be the first range table entry.
+	 * However, when it is a view, a new RTE is added. Original RTE
+	 * stays at the first place. This type of queries are not supported.
+	 */
+	if (queryTree->resultRelation != 1)
```

Comment:
I really don't want to hardcode that assumption. Which turns out not to actually be true in general...